frontend/packages/client/src/components/Community/StrategyEditorModal/index.js

```diff
@@ -0,0 +1,155 @@
+import React, { useEffect, useState } from 'react';
+import networks from 'networks';
+import StrategySelector from './StrategySelector';
+import StrategyInformationForm from './StrategyInformationForm';
+import { ActionButton } from 'components';
+import { isValidAddress } from 'utils';
+
+const networkConfig = networks[process.env.REACT_APP_FLOW_ENV];
+
+const ModalSteps = {
+  1: 'select-strategy',
+  2: 'strategy-information',
+};
+
+const initialFormFields = {
+  contractAddress: '',
+  contractName: '',
+  maxWeight: '',
+  minimunBalance: '',
+};
+
+const formFields = Object.keys(initialFormFields);
+
+export default function StrategyEditorModal({
+  strategies = [],
+  enableDismiss = true,
+  onDismiss = () => {},
+  // callback to pass data collected and closed modal
+  onDone = () => {},
+} = {}) {
+  // handles two steps inside modal
+  // 1 - strategy name
+  // 2 - contract strategy info
+  const [step, setSep] = useState(ModalSteps[1]);
+  const [formIsValid, setIsFormValid] = useState(false);
+
+  const [strategyData, setStrategyData] = useState({
+    name: '',
+    contract: { ...initialFormFields },
+  });
+
+  // this useEffect validates form on second step
+  useEffect(() => {
+    const requiredFields = {
+      contractAddress: (addr) =>
+        addr?.trim().length > 0 && isValidAddress(addr),
+      contractName: (name) =>
+        name?.trim().length > 0 && name?.trim().length <= 150,
+      maxWeight: (maxWeight) =>
+        maxWeight?.trim().length > 0 && /^[0-9]+$/.test(maxWeight),
+      minimunBalance: (minimunBalance) =>
+        minimunBalance?.trim().length > 0 && /^[0-9]+$/.test(minimunBalance),
+    };
+    const isValid = Object.keys(requiredFields).every((field) =>
+      requiredFields[field](strategyData.contract[field])
+    );
+    setIsFormValid(isValid);
+  }, [strategyData]);
+
+  // user selected strategy move to second step to enter information
+  const setStrategy = (strategyName) => {
+    //
+    // STRATEGY CONFIGURATION
+    //
+    // If strategy selected is 'one-address-one-vote'
+    // then no more information is required(FE uses strategy configuration)
+    // modal should be closed and strategy should be ready to be added
+    if (strategyName === 'one-address-one-vote') {
+      const strategyConfig =
+        networkConfig.strategiesConfig['one-address-one-vote'];
```

Comment:
backend needs this information for `one-address-one-vote` sent even though user does not enter the data manually @rvmelkonian is aware of this payload and tested backend and works as expected

Reply:
yup, can confirm this. We can write some additional code on the server to change this if we want. But priority was to get this implementation out the door.